zfs-2.1.3/module/zfs/arc.c:1208:17: error: unused variable ‘hdr’

<!-- Please fill out the following template, which will help other contributors address your issue. -->

Compiling on Centos 8 Stream with latest elrep kernel-ml  5.16.13-1.el8.elrepo.x86_64 fails

```
Making all in module
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-4)
  You are using:           gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-10)
/root/zfs-2.1.3/module/zfs/arc.c: In function ‘hdr_l2only_dest’:
/root/zfs-2.1.3/module/zfs/arc.c:1208:17: error: unused variable ‘hdr’ [-Werror=unused-variable]
  arc_buf_hdr_t *hdr = vbuf;
                 ^~~
cc1: all warnings being treated as errors
make[5]: *** [scripts/Makefile.build:287: /root/zfs-2.1.3/module/zfs/arc.o] Error 1
make[4]: *** [scripts/Makefile.build:549: /root/zfs-2.1.3/module/zfs] Error 2
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [Makefile:1846: /root/zfs-2.1.3/module] Error 2
make[2]: *** [Makefile:55: modules-Linux] Error 2
make[1]: *** [Makefile:920: all-recursive] Error 1
```

### System information
Distribution Name	| Centos-Stream
Distribution Version	| 8
Kernel Version	| 5.16.13-1.el8.elrepo.x86_64
Architecture	| x886_64
OpenZFS Version	| 2.1.3

tried your patch @nabijaczleweli  but 
```
../../module/zfs/arc.c: In function 'buf_dest':
../../module/zfs/arc.c:1220:17: error: 'buf' undeclared (first use in this function); did you mean 'vbuf'?
  mutex_destroy(&buf->b_evict_lock);
                 ^~~
                 vbuf
../../module/zfs/arc.c:1220:17: note: each undeclared identifier is reported only once for each function it appears in
  CC       bqueue.lo
  CC       cityhash.lo
make[5]: *** [Makefile:1336: arc.lo] Error 1
make[5]: *** Waiting for unfinished jobs....
```

--- a/module/zfs/arc.c
+++ b/module/zfs/arc.c
@@ -1204,11 +1204,11 @@ hdr_full_dest(void *vbuf, void *unused)
 static void
 hdr_full_crypt_dest(void *vbuf, void *unused)
 {
-	(void) unused;
-	arc_buf_hdr_t *hdr = vbuf;
+	(void) vbuf, (void) unused;
 
 	hdr_full_dest(vbuf, unused);
-	arc_space_return(sizeof (hdr->b_crypt_hdr), ARC_SPACE_HDRS);
+	arc_space_return(sizeof (((arc_buf_hdr_t *)NULL)->b_crypt_hdr),
+	    ARC_SPACE_HDRS);
 }
 
 static void

